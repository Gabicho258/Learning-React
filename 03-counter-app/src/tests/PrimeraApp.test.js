// import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import React from 'react'
import "@testing-library/jest-dom"
import {shallow} from "enzyme"


describe('Pruebas en PrimeraApp', () => {
    // test('Debe mostrar el mensaje "Hola soy yo"', () => {
    //     const saludo = "Hola soy yo";
    //     const {getByText} = render(<PrimeraApp saludo = {saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })
    test('Debe mostrar primera app correctamente', () => {
        const saludo = "Hola soy yo"
        const wrapper = shallow(<PrimeraApp saludo = {saludo}/>)
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe mostrar el subtitulo enviado por promps', () => {
        const saludo = "Hola soy yo"
        const subtitulo = "Soy un subtituloasd"
        const wrapper = shallow(
            <PrimeraApp 
                saludo = {saludo}
                subtitulo = {subtitulo}
            />
        )
        const textoParrafo = wrapper.find('p').text();
        console.log(textoParrafo);
        expect(textoParrafo).toBe(subtitulo);
    })
    
    
})
