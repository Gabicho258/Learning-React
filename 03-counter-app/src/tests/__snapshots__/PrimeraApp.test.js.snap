// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp Debe mostrar primera app correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy yo
  </h1>
  <p>
    Soy un subtitulo   
  </p>
</Fragment>
`;
